lib/experimental/site-editor-permalinks.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+add_action(
+	'block_editor_settings_all',
+	function ( $settings ) {
+		$settings['__experimentalDashboardLink'] = admin_url();
+		return $settings;
+	}
+);
+
+function gutenberg_remove_query_args( $args ) {
+	$query_string = $_SERVER['QUERY_STRING'];
+	$query        = wp_parse_args( $query_string );
+	foreach ( $args as $arg_name ) {
+		unset( $query[ $arg_name ] );
+	}
+	return $query;
+}
+
+function gutenberg_get_site_editor_url( $path = '', $query = array() ) {
+	$query_string = build_query( array_merge( $query, $path ? array( 'p' => $path ) : array() ) );
+	$base_url     = admin_url( 'site-editor.php' );
+	return $query_string ? $base_url . '?' . $query_string : $base_url;
+}
+
+function gutenberg_get_posts_dataviews_url( $path = '', $query = array() ) {
+	$query_string = build_query(
+		array_merge(
+			$query,
+			$path ? array(
+				'page' => 'gutenberg-posts-dashboard',
+				'p'    => $path,
+			) : array( 'page' => 'gutenberg-posts-dashboard' )
+		)
+	);
+	$base_url     = admin_url( 'admin.php' );
+	return $query_string ? $base_url . '?' . $query_string : $base_url;
+}
+
+function gutenberg_redirect_site_editor_to_design() {
+	global $pagenow;
+	if ( 'site-editor.php' !== $pagenow || isset( $_REQUEST['p'] ) || ! isset( $_SERVER['querystring'] ) ) {
```

Comment:
The `isset( $_SERVER['querystring'] )` is always false for me (local `wp-env` Docker WordPress). I had to delete it to make the redirects work.